tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ConfidentialClientIntegrationTests.cs

```diff
@@ -628,7 +634,66 @@ private async Task RunOnBehalfOfTestAsync(LabResponse labResponse)
                 .WithAuthority(new Uri(oboHost + authResult.TenantId), true)
                 .WithClientSecret(secret)
                 .WithTestLogging()
-                .Build();
+                .BuildConcrete();
+
+            var userCacheRecorder = confidentialApp.UserTokenCache.RecordAccess();
+
+            UserAssertion userAssertion = new UserAssertion(authResult.AccessToken);
+
+            string atHash = userAssertion.AssertionHash;
+
+            authResult = await confidentialApp.AcquireTokenOnBehalfOf(s_scopes, userAssertion)
+                .ExecuteAsync(CancellationToken.None)
+                .ConfigureAwait(false);
+
+            MsalAssert.AssertAuthResult(authResult, user);
+            Assert.AreEqual(atHash, userCacheRecorder.LastAfterAccessNotificationArgs.SuggestedCacheKey);
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            await confidentialApp.GetAccountsAsync().ConfigureAwait(false);
+#pragma warning restore CS0618 // Type or member is obsolete
+            Assert.IsNull(userCacheRecorder.LastAfterAccessNotificationArgs.SuggestedCacheKey);
```

Comment:
what is this test testing?

Reply:
see https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2667/files#r648732899

Reply:
sorry, what is it testing specifically? that link just goes to this PR.

Reply:
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2667#discussion_r648732899

Wrong link